@uportal/notification-list/src/components/NotificationList.vue

```diff
@@ -1,21 +1,164 @@
+import queryString from 'query-string';
+
+/*eslint-disable */
 
 export default {
     name: 'notification-list',
     data() {
-        return {};
+        return {
+            notifications: [],
+            selectedAction: null,
+            loaded: false
+        };
+    },
+    props: {
+        oidcUrl: {
+            type: String,
+            default: '/uPortal/api/v5-1/userinfo'
+        },
+        notificationApiUrl: {
+            type: String,
+            default: '/NotificationPortlet/api/v2/notifications'
+        },
+        debug: {
+            type: Boolean,
+            default: false
+        },
+        colorMap: {
+            type: Object,
+            default: () => ({
+                Announcement: '#6649bb',
+                Holds: '#487df9',
+                'To-Do': '#c85a89'
+            })
+        }
+    },
+    methods: {
+        gotoAction(action) {
+            let form = document.createElement('form');
+            form.action = action.apiUrl;
+            form.method = 'POST';
+            form.target = '_blank';
+            form.style.display = 'none';
+            document.body.appendChild(form);
+            form.submit();
+        },
+        markAllAsRead() {
+            this.gotoAction(this.markAllAsReadLink);
+        },
+        handleOidcError(err) {
+            this.hasError = true;
+            this.errorMessage = 'There was a problem authorizing this request.';
+        },
+        handleWflError(err) {
+            this.hasError = true;
+            this.errorMessage = 'There was a problem retrieving notifications.';
+        },
+        async getToken() {
+            try {
+                return await oidc({ userInfoApiUrl: this.oidcUrl });
+            } catch (err) {
+                console.error(err);
+                this.handleOidcError(err);
+            }
+        },
+        async fetchNotificationData() {
+            const token = this.debug ? null : (await this.getToken()).encoded;
+
+            try {
+                const response = await fetch(this.notificationApiUrl, {
+                    credentials: 'same-origin',
+                    headers: {
+                        Authorization: 'Bearer ' + token,
+                        'content-type': 'application/jwt'
+                    }
+                });
+
+                if (!response.ok || response.status !== 200) {
+                    throw new Error(response.statusText);
+                }
+
+                const payload = await response.json();
+
+                this.notifications = payload;
+            } catch (err) {
+                this.err = err;
+            } finally {
+                this.loaded = true;
+            }
+        }
     },
-    computed: {}
+    mounted() {
+        this.fetchNotificationData();
+
+        const queryParams = queryString.parse(location.search);
+        this.highlight = queryParams.highlight;
+    },
+    filters: {
+        dueDate: function(value) {
```

Comment:
:art: 

```suggestion
        dueDate (value) {
```

Reply:
function is unnecessary in this component now, removed